karapace/schema_reader.py

```diff
@@ -487,7 +487,7 @@ def _handle_msg_config(self, key: dict, value: dict | None) -> None:
     def _handle_msg_delete_subject(self, key: dict, value: dict | None) -> None:  # pylint: disable=unused-argument
         if value is None:
             LOG.warning("DELETE_SUBJECT record doesnt have a value, should have")
-            return
+            raise ValueError("DELETE_SUBJECT record doesnt have a value, should have")
```

Comment:
mmm what happens if karapace its running and we throw an exception? Its going to make the process fail? Often happens that other users/programs insert random data in the topic, would it mean that we have to cleanup the topic to fix that scenario?

Reply:
ah I see with the new flag this will "only" skip the message

Reply:
Can we add a unit test verifying that?

Reply:
Not sure if its collected and managed

Reply:
> mmm what happens if karapace its running and we throw an exception? Its going to make the process fail? Often happens that other users/programs insert random data in the topic, would it mean that we have to cleanup the topic to fix that scenario?

Yes I get the idea, the things is that we will introduce the config to allow users to toggle the behaviour from shutdown to continue processing the records. 
If they set the config `schema_reader_strict_mode` to `True` then their service will always shutdown whenever we read the erroring schema record.

Reply:
> Can we add a unit test verifying that?

It is already tested, if you check this PR (#936), there I didn't make any changes to the existing tests which already test this behaviour in question, that the message processing will reach the end and the offset is reflected. See this test - https://github.com/Aiven-Open/karapace/blob/main/tests/unit/test_schema_reader.py#L197 if it makes sense.